csharp/src/Google.Protobuf/UnknownFieldSet.cs

```diff
@@ -231,7 +244,65 @@ private bool MergeFieldFrom(CodedInputStream input)
                 default:
                     throw InvalidProtocolBufferException.InvalidWireType();
             }
-        }
+        }
+
+#if GOOGLE_PROTOBUF_SUPPORT_SPAN
+        /// <summary>
+        /// Parse a single field from <paramref name="input"/> and merge it
+        /// into this set.
+        /// </summary>
+        /// <param name="input">The coded input reader containing the field</param>
+        /// <returns>false if the tag is an "end group" tag, true otherwise</returns>
+        private bool MergeFieldFrom(ref CodedInputReader input)

```

Comment:
I'm wondering if there's a way to avoid code duplication between MergeFieldFrom(ref CodedInputReader input) and MergeFieldFrom(CodedInputStream input).   It would be nice if whenever GOOGLE_PROTOBUF_SUPPORT_SPAN is enabled, all the MergeFieldFrom invocations would actually invoke  MergeFieldFrom(ref CodedInputReader input) internally because that would make all the existing tests could be used to test the new implementation for free.

Reply:
I don't think so. Just because GOOGLE_PROTOBUF_SUPPORT_SPAN is present doesn't necessarily mean that there is code generated to use.